### ISSUE TITLE
Bug: Toolbar does not appear if offline or YouTube is blocked

### ISSUE THREAD

**Issue by chrisgurney** [andrewmcgivery/obsidian-soundscapes#26]
**Problem**

On launch of Obsidian, my Soundscapes toolbar controls are no longer loading this afternoon, while they were working fine this morning.

![CleanShot 2024-02-28 at 14 09 40](https://github.com/andrewmcgivery/obsidian-soundscapes/assets/17337703/44aa1b19-ffeb-4459-8338-554e61b018d1)

My hunch right now is that I am using an app on MacOS called [SelfControl](https://selfcontrolapp.com/) that completely blacklists a bunch of websites. One I have set is YouTube (`www.youtube.com`). ~~At the moment I can't disable my block (that's by design) to see if that's the issue. I'll re-confirm this evening and update my post then. Could not having access to YouTube (say using Obsidian completely offline) prevent Soundscapes from starting up correctly?~~ (Confirmed, see next comment.)

![CleanShot 2024-02-28 at 14 01 58@2x](https://github.com/andrewmcgivery/obsidian-soundscapes/assets/17337703/d38f554b-b0b3-4e74-855b-61982e0bcf14)

![CleanShot 2024-02-28 at 14 42 58@2x](https://github.com/andrewmcgivery/obsidian-soundscapes/assets/17337703/f330fd7e-fbc6-4a26-ac13-b9bf94eef04f)

**Other things I've tried...**
 
- I have some CSS in place to style the Soundscapes portion of the toolbar, but I disabled/removed it and ruled it out as the problem.

**Comment by chrisgurney** [comment]
@andrewmcgivery I've confirmed that the Soundscapes toolbar does _not_ appear in Obsidian, AND the music player does not work if:

- The user is offline when launching Obsidian; OR
- YouTube is blocked (via an app like [SelfControl](https://selfcontrolapp.com/)) when launching Obsidian.

Going offline and blocking sites are both strategies I use when I want to focus and minimize distractions, which seem analogous with the goals of this plugin.

Is it possible to update the plugin to support offline use?

Thanks!

**Comment by chrisgurney** [comment]
I don't know for sure but iIt looks like the player is created [here](https://github.com/andrewmcgivery/obsidian-soundscapes/blob/9f668998d1942e12256d6be414f1982be24a37e5/main.ts#L318)... but it's _within_ the callback from YouTube's API.

I'm not sure where the change is to be made; perhaps in `main.ts` `createPlayer()`?

I think then if the Soundscape is set to _My Music_ the player can just go ahead and be initiated:
```
if (this.soundscapeType === SOUNDSCAPE_TYPE.MY_MUSIC) {
   this.onPlayerReady.bind(this)
```

**Comment by andrewmcgivery** [comment]
Apologies, I was out on a work trip. :)

I've added this to my TODO list to take a look at. :)

**Comment by andrewmcgivery** [comment]
This should now be fixed in 1.3.0 (just released). :)

**Comment by chrisgurney** [comment]
@andrewmcgivery Much appreciated, thank you!
